core/math/vector2.h

```diff
@@ -243,12 +240,11 @@ _FORCE_INLINE_ Vector2 Vector2::operator-() const {
 	return Vector2(-x, -y);
 }
 
-_FORCE_INLINE_ bool Vector2::operator==(const Vector2 &p_vec2) const {
-	return x == p_vec2.x && y == p_vec2.y;
-}
-
-_FORCE_INLINE_ bool Vector2::operator!=(const Vector2 &p_vec2) const {
-	return x != p_vec2.x || y != p_vec2.y;
+std::partial_ordering Vector2::operator<=>(const Vector2 &p_other) const {
+	if (x <=> p_other.x == 0) {
```

Comment:
```suggestion
	if ((x <=> p_other.x) == 0) {
```
Generally I think precedence should be just left without parentheses but this operator is still pretty niche so I think this helps readability here, for all

Reply:
Come to think of it, this use of the three-way operator was redundant. I can just change it to the functionally-equivalent:
```suggestion
	if (x == p_other.x) {
```

Reply:
Absolutely less confusing!